### PR TITLE
✨ BigTreeTech Manta Series (M4P, M5P, M8P V1.0 & V1.1, E3 EZ)

### PR DESCRIPTION
**Pull request by thisiskeithb**
### Description

Add support for the BigTreeTech Manta Series:
- [Manta M4P](https://github.com/bigtreetech/Manta-M4P)
- [Manta M5P](https://github.com/bigtreetech/Manta-M5P)
- [Manta M8P V1.0 & V1.1](https://github.com/bigtreetech/Manta-M8P)
- [Manta E3 EZ](https://github.com/bigtreetech/Manta-E3EZ)

#### M4P
- Onboard SD (MCU) is not accessible by Marlin & inserting an SD card causes the board to reboot. I tried tracking this down with `M43`, but it was problematic:
- `M43 W` hangs on Pin 57 / `PF0` / `PF_0` which is labeled as `OSC_IN` in [the schematic](https://github.com/bigtreetech/Manta-M4P/blob/master/Hardware/bigtreetech_manta_m4p_v2.1_220608_SCH.pdf).
- LCD contrast is not properly initialized unless you set `SDCARD_CONNECTION` to LCD. It's probably related to ONBOARD not working and this is *probably* a bug within Marlin getting stuck on not being able to access the onboard SD & skipping proper contrast initialization...maybe? I couldn't track it down.

#### M5P V1.0

- Nothing to report as I don't have this board to test with.

#### M8P V1.0
- Several common LCD types did not work correctly / SD was inaccessible on the LCD.

#### M8P V1.1

- Nothing to report as I don't have this board to test with.

#### E3 EZ
- It's not so much an issue, but an FYI: The external USB port is connected to the CB1/CM4 and not the MCU, so you can't use an external host to control the printer. Marlin will run fine without a CB1/CM4, so that hardware is completely optional.

### CODE REVIEW

**Comment by ils15** [comment]
klipper board for marlin firmware
its strange...

**Comment by thisiskeithb** [comment]
> its strange...

Most motherboards can run multiple firmwares. It's not strange at all.

**Comment by thisiskeithb** [comment]
M8P V1.1 was a late add since I just found out about it & there's also a Manta M5P on the way, but don't have any information about that board yet.

**Comment by Pseudo-Nym** [comment]
Looking forward to testing this out on my M8P v1.0

**Comment by mriscoc** [comment]
It seems this broke the BTT SKR Mini E3 V3 support for Ender3V2 DWIN/DACAI displays.

**Comment by thisiskeithb** [comment]
> It seems this broke the BTT SKR Mini E3 V3 support for Ender3V2 DWIN/DACAI displays.

I’m guessing it was the variant changes in https://github.com/MarlinFirmware/Marlin/pull/25001/commits/b4b6856b506e24a23fe087f90c7c55672910f746.

I tested BTT’s TFTs on the SKR Mini E3 V3 & Manta E3 EZ, but that’s as far as I went.

**Comment by thinkyhead** [comment]
> > It seems this broke the BTT SKR Mini E3 V3 support for Ender3V2 DWIN/DACAI displays.
> 
> I’m guessing it was the variant changes in [b4b6856](https://github.com/MarlinFirmware/Marlin/commit/b4b6856b506e24a23fe087f90c7c55672910f746).
> 
> I tested BTT’s TFTs on the SKR Mini E3 V3 & Manta E3 EZ, but that’s as far as I went.

We must be cautious when adding support for a new board not to change things that will break other boards. In many cases we are setting `PIN_SERIAL*_RX` as an environment build flag to override the variant, which avoids changing the variant itself and breaking other boards that use it.

So let's revert changes to the variant and move the needed settings to an environment that applies specifically to the targeted board, so that previously-existing boards are working the same way as before and are not touched by the Manta-specific changes.

Comments from BigTreeTech suggest that we should be able to eliminate some or all the custom variants that were derived from the standard variants and replace them with build flags. Let's make a concerted effort to get all the boards, environments, and variants sorted out so that they are as standardized as possible, and where possible get rid of custom variants and use the standard ones, starting with all the BTT boards.

**Comment by thisiskeithb** [comment]
> In many cases we are setting `PIN_SERIAL*_RX` as an environment build flag to override the variant, which avoids changing the variant itself and breaking other boards that use it.

I think you know as well as most of us that it’s impossible to test every single variation/combination of hardware 🙂

These all use the same build environment (`STM32G0B1RE_btt`), so the option was to change the environment or change the variant.

It might make sense to revert the variant changes and setup a new `STM32G0B1RE_btt` environment for the G0-based Manta boards.

**Comment by thisiskeithb** [comment]
See https://github.com/MarlinFirmware/Marlin/pull/25431

**Comment by pdsavas** [comment]
I need some help. I followed the instructions in the user manual to enable the USB ports with a CM4. I've added the dtoverlay line to the config.txt file on the boot partition, but the ports are still not working. Is there something else that needs to be done? I've asked on the BTT discord, but get no answer there.

**Comment by thisiskeithb** [comment]
@pdsavas This Issue Queue is for Marlin bug reports and development-related issues, and we prefer not to handle user-support questions here. (As noted on [this page](https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.1.x/.github/contributing.md#i-dont-want-to-read-this-whole-thing-i-just-have-a-question).) For best results getting help with configuration and troubleshooting, please use the following resources:

- [Marlin Documentation](https://marlinfw.org) hosted on MarlinFW.org
- [Marlin Forum](https://forums.reprap.org/list.php?415) hosted on RepRap.org
- [Marlin Firmware](https://www.facebook.com/groups/1049718498464482/) Facebook Group 
- [Marlin Firmware for 3D Printers](https://www.facebook.com/groups/3Dtechtalk/) Facebook Group
- [Marlin Configuration](https://www.youtube.com/results?search_query=marlin+configuration) on YouTube
- [Marlin Firmware](https://discord.gg/n5NJ59y) Discord server

After seeking help from the community, if the consensus points to a bug in Marlin, then you should post a [bug report](https://github.com/MarlinFirmware/Marlin/issues/new/choose).

**Comment by pdsavas** [comment]
Sorry for the trouble.

File: buildroot/share/PlatformIO/variants/MARLIN_G0B1RE/variant_MARLIN_STM32G0B1RE.cpp [hunk 1]
```diff
@@ -130,14 +130,7 @@ WEAK void SystemClock_Config(void)
...
   // Reset clock registers (in case bootloader has changed them)
-  RCC->CR |= RCC_CR_HSION;
-  while (!(RCC->CR & RCC_CR_HSIRDY))
-      ;
-  RCC->CFGR = 0x00000000;
-  RCC->CR = RCC_CR_HSION;
-  while (RCC->CR & RCC_CR_PLLRDY)
-      ;
-  RCC->PLLCFGR = 0x00001000;
+  SystemInit();
```
Comment (mistic100) [inline buildroot/share/PlatformIO/variants/MARLIN_G0B1RE/variant_MARLIN_STM32G0B1RE.cpp:133]: @thisiskeithb I recently updated my SKR Mini E3 v3.0 to the lastest bugfix 2.1 and this change made the printer irresponsive, I just get a blank screen after boot

I have no idea what this code does, but after reverting to the previous version (keeping all other changed), it booted again.

Comment (thisiskeithb) [reply buildroot/share/PlatformIO/variants/MARLIN_G0B1RE/variant_MARLIN_STM32G0B1RE.cpp:133]: > @thisiskeithb I recently updated my SKR Mini E3 v3.0 to the lastest bugfix 2.1 and this change made the printer irresponsive, I just get a blank screen after boot
> 
> I have no idea what this code does, but after reverting to the previous version (keeping all other changed), it booted again.

@mistic100: Please create a proper [Bug Report](https://github.com/MarlinFirmware/Marlin/issues/new/choose) and be sure to attach your configs.

I tested this PR on an SKR Mini E3 V3 and did not have any issues.

**Pull request merged**